src/vs/editor/browser/viewParts/lines/viewLines.ts

```diff
@@ -718,6 +731,10 @@ export class ViewLines extends ViewPart implements IVisibleLinesHost<ViewLine>,
 			newScrollTop = this._computeMinimumScrolling(viewportStartY, viewportEndY, boxStartY, boxEndY, verticalType === viewEvents.VerticalRevealType.Top, verticalType === viewEvents.VerticalRevealType.Bottom);
 		}
 
+		const scrollTopLineNumber = Math.ceil(newScrollTop / this._lineHeight);
+		if (this._stickyScrollEnabled && range && range?.startLineNumber - scrollTopLineNumber <= this._maxNumberStickyLines) {
+			return newScrollTop + this._lineHeight * (range?.startLineNumber - scrollTopLineNumber - this._maxNumberStickyLines - 1);
+		}
```

Comment:
Could we remove this logic from here and adjust `boxStartY` above in the same area where `cursorSurroundingLines` is handled? The code above needs to be tweaked, maybe we introduce two variables, `paddingTop` and `paddingBottom` and lines 681-697 can be modified to define `paddingTop`/`paddingBottom` and then those can be subtracted/added to `boxStartY`/`boxEndY`.